src/orchard/xref.clj

```diff
@@ -13,21 +15,47 @@
     (symbol? thing) (var-get (find-var thing))
     (fn? thing) thing))
 
-(defn fn-deps
-  "Returns a set with all the functions invoked by `val`.
-  `val` can be a function value, a var or a symbol."
-  {:added "0.5"}
-  [val]
-  (let [val (as-val val)]
-    (set (some->> val class .getDeclaredFields
+(defn- fn-name [^java.lang.Class f]
+  (-> f .getName repl/demunge symbol))
+
+(defn fn-deps-class
+  "Returns a set with all the functions invoked by `v`.
+  `v` can be a function class or a symbol."
+  {:added "0.8"}
+  [v]
+  (let [^java.lang.Class v (if (class? v)
+                             v
+                             (eval v))]
+    (set (some->> v .getDeclaredFields
                   (keep (fn [^java.lang.reflect.Field f]
                           (or (and (identical? clojure.lang.Var (.getType f))
                                    (java.lang.reflect.Modifier/isPublic (.getModifiers f))
                                    (java.lang.reflect.Modifier/isStatic (.getModifiers f))
                                    (-> f .getName (.startsWith "const__"))
-                                   (.get f val))
+                                   (.get f (fn-name v)))
                               nil)))))))
 
+(def ^java.lang.reflect.Field classCache
+  (let [classCache* (.getDeclaredField clojure.lang.DynamicClassLoader "classCache")]
+    (.setAccessible classCache* true)
+    (.get classCache* clojure.lang.DynamicClassLoader)))
```

Comment:
I've managed to fix the IllegalAccess issue. This was something I tried before, not sure why it's working now, but I'm happy.

Reply:
Just add some comments explaining this for posterity's sake.